### ISSUE TITLE
editor.action.jumpToBracket should move the cursor to the closest enclosing bracket when not on a bracket character

### ISSUE THREAD

**Issue by Tyriar** [microsoft/vscode#1772]
**Steps to reproduce**
1. Create a js file with the following contents:
   
   ``` js
   var a = {
     "foo": "bar"
   };
   ```
2. Put the cursor after `"foo"`
3. Press <kbd>ctrl</kbd>+<kbd>shift</kbd>+<kbd>]</kbd> to trigger `editor.action.jumpToBracket`

**Expected**

The cursor jumps to either the opening `{` (Atom does this) or closing `}` (Sublime does this).

**Actual**

The cursor does not move.

**Notes**

Sublime keeps track of and underlines the closest set of brackets constantly, not just when the cursor is on the bracket.

**Comment by alexandrudima** [comment]
This is a great idea! :+1:

**Comment by Dlorwisdom** [comment]
I too would like to see this feature!! I've spent the last hour trying to figure out why my VSCode wasn't behaving properly, but know I know that it is!

**Comment by kossel** [comment]
I want this feature too! I even created a stackoverflow question to know that it doesn't exist yet.

**Comment by Tyriar** [comment]
Bumping this considering I'm testing the new [Atom keymap extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode.atom-keybindings) in https://github.com/Microsoft/vscode/issues/14318

This was one of my annoyances being a previous Atom user. It makes the command far more usable, I used to use it all the time but since VS Code doesn't scan back to the previous bracket, I very rarely use it.

**Comment by illz** [comment]
I'd also love this

**Comment by abhijithda** [comment]
In 'vim' it's pressing '[' character twice to jump to the beginning of the braces that starts at column 0 (which usually the beginning of the function in our case). It would be great to see similar stuffs...

```
def fun 
{
 ...
}
```

**Comment by SailorMax** [comment]
@Tyriar Looks like VSCode 1.16.1 already move cursor between brackets...
But it not scroll to new cursor position :) In result in cases when opposite bracket at few screens upper or downer, user have to move cursor by left/right arrows to see new position of cursor.
Is it separate bug?

**Comment by Tyriar** [comment]
@SailorMax the scrolling appears fixed in Insiders

**Comment by sbatten** [comment]
Seems this is working and should be closed.

**Comment by Tyriar** [comment]
Well the behavior that we do is different to what I described, we appear to scan forward for the next bracket, it is much more useful to go up to the bracket containing the cursor. @rebornix do you know who added the (new) existing behavior?

**Comment by Tyriar** [comment]
@sbatten 

Before:

<img width="360" alt="screen shot 2018-09-21 at 10 43 50 am" src="https://user-images.githubusercontent.com/2193314/45897193-550fc200-bd8b-11e8-9ef4-e4a6d796fc65.png">

After 2 times actual:

<img width="285" alt="screen shot 2018-09-21 at 10 44 33 am" src="https://user-images.githubusercontent.com/2193314/45897241-6e187300-bd8b-11e8-8dde-04b23c3d460b.png">

After 2 times expected:

<img width="322" alt="screen shot 2018-09-21 at 10 44 58 am" src="https://user-images.githubusercontent.com/2193314/45897258-7b356200-bd8b-11e8-91ac-84374f28a06e.png">

In other words, we should scan backwards when the cursor is not on a bracket, not forwards.

**Comment by sbatten** [comment]
@Tyriar makes sense, thanks for clarifying

**Comment by szabi** [comment]
@Tyriar, @sbatten

If the behaviour would jump just right _out_ of the bracket then presssing it twice would work as expected with scanforward.

Also, this might be very much dependent on what you are used to, but I personally find it more often  to wish to jump just right out of the bracket. When typing an opening bracket a closing bracket is inserted after the cursor. I start typing what I want inside the bracket. Then I want to continue outside of it, ideally without lifting my fingers to the cursor keys, ie. with a Modifier+key shortcut.

**Comment by alexandrudima** [comment]
For verification:
```ts
const x = {
  something: [1, 2, 3],
  another: "true",
  interesting: [4, 5, 6]
}
```

Place the cursor in various spots in the above code snippet and check that the encompassing brackets are jumped to when pressing `` ctrl+shift+\ ``, which might not be necessarily the "closest" brackets.
